wantlocker/user-posts-code-challenge
# Create prisma schema

### Description
- Add dev setup
  - Add `docker-compose.yml` file which spins up a Postgres 12 container
  - Add `.tool-versions` file for asdf to specify nodejs version
    - My laptop didn't have the version necessary to run Prisma, so I added this. It's also nice to standardize versions used across the dev team
- Create `User` and `Post` models in the Prisma schema
  - Table names are plural, lowercase
  - Map Prisma field names to snake-cased column names
  - Create minor indexing - we can add more once optimization is required
- Ran `npx prisma migrate dev --name create_users_and_posts_tables` to generate the tables using the dockerized Postgres DB

### Test plan
- Start the Postgres container
```
// Run at /api level directory
docker-compose up -d
```
- Run `npx prisma migrate dev`
- Verify migration happens successfully
